core/model/modx/modx.class.php

```diff
@@ -1583,6 +1583,7 @@ public function invokeEvent($eventName, array $params= array ()) {
             //$this->log(modX::LOG_LEVEL_DEBUG,'System event '.$eventName.' was executed but does not exist.');
             return false;
         }
+        $oldEventName = $this->event->name;//store current eventname        
```

Comment:
Isn't there a case where `$modx->event` is not yet an object ? (i guess before the first event is fired)